cszczepaniak/cribbly-frontend
# Create landing page

# What I added (link any issues addressed)
- A very basic landing page
- A very basic welcome page
- The plumbing to connect auth in with everything. It sorta works

# How to test it
- Run the react app and go to localhost:3000 in your browser
- Click the button, sign in, you should see a welcome page greeting you (with your name) and a logout button
- Click the logout button; you should be returned to the landing page
- While logged out, try to navigate to localhost:3000/home. You should be directed to Auth0's login page before being allowed to proceed.
  - When you do this, the landing page flashes briefly on screen before redirecting you to the home page. I don't like it...

# Some links I used
- https://community.auth0.com/t/persisting-login-between-refreshes/22675
- https://github.com/auth0/auth0-react/blob/master/EXAMPLES.md#1-protecting-a-route-in-a-react-router-dom-app

# What we'll still need to solve
- If you refresh on a protected route (like /home), you'll need to log back in every time. I want to either store a token in a cookie or look into refresh tokens to make that not necessary
- The frontend has to talk to the backend on login to store a user in the database, or get the user if they already exist in the database. I think we'd use a `useEffect` hook on the home page to handle that.
